<!DOCTYPE html>
<html lang='en'>
<head>
<title>musl-sandbox.patch\chromium\community - aports - Unnamed repository; edit this file 'description' to name the repository.
</title>
<meta name='generator' content='cgit v1.2.1'/>
<meta name='robots' content='index, nofollow'/>
<link rel='stylesheet' type='text/css' href='/cgit.css'/>
<link rel='shortcut icon' href='//alpinelinux.org/alpine-logo.ico'/>
<link rel='alternate' title='Atom feed' href='http://git.alpinelinux.org/aports/atom/community/chromium/musl-sandbox.patch?h=master' type='application/atom+xml'/>
<link rel='vcs-git' href='git://git.alpinelinux.org/aports' title='aports Git repository'/>
<link rel='vcs-git' href='https://git.alpinelinux.org/aports' title='aports Git repository'/>
</head>
<body>
<div id='cgit'><table id='header'>
<tr>
<td class='logo' rowspan='2'><a href='/'><img src='//wiki.alpinelinux.org/images/alogo.png' alt='cgit logo'/></a></td>
<td class='main'><a href='/'>index</a> : <a title='aports' href='/aports/'>aports</a></td><td class='form'><form method='get'>
<select name='h' onchange='this.form.submit();'>
<option value='1.10-stable'>1.10-stable</option>
<option value='1.9'>1.9</option>
<option value='2.0-stable'>2.0-stable</option>
<option value='2.1-stable'>2.1-stable</option>
<option value='2.2-stable'>2.2-stable</option>
<option value='2.3-stable'>2.3-stable</option>
<option value='2.4-stable'>2.4-stable</option>
<option value='2.5-stable'>2.5-stable</option>
<option value='2.6-stable'>2.6-stable</option>
<option value='2.7-stable'>2.7-stable</option>
<option value='3.0-stable'>3.0-stable</option>
<option value='3.1-stable'>3.1-stable</option>
<option value='3.10-stable'>3.10-stable</option>
<option value='3.11-stable'>3.11-stable</option>
<option value='3.12-stable'>3.12-stable</option>
<option value='3.2-stable'>3.2-stable</option>
<option value='3.3-stable'>3.3-stable</option>
<option value='3.4-stable'>3.4-stable</option>
<option value='3.5-stable'>3.5-stable</option>
<option value='3.6-stable'>3.6-stable</option>
<option value='3.7-stable'>3.7-stable</option>
<option value='3.8-stable'>3.8-stable</option>
<option value='3.9-stable'>3.9-stable</option>
<option value='master' selected='selected'>master</option>
</select> <input type='submit' value='switch'/></form></td></tr>
<tr><td class='sub'>Unnamed repository; edit this file 'description' to name the repository.
</td><td class='sub right'>uwsgi</td></tr></table>
<table class='tabs'><tr><td>
<a href='/aports/about/'>about</a><a href='/aports/'>summary</a><a href='/aports/refs/'>refs</a><a href='/aports/log/community/chromium/musl-sandbox.patch'>log</a><a class='active' href='/aports/tree/community/chromium/musl-sandbox.patch'>tree</a><a href='/aports/commit/community/chromium/musl-sandbox.patch'>commit</a><a href='/aports/diff/community/chromium/musl-sandbox.patch'>diff</a><a href='/aports/stats/community/chromium/musl-sandbox.patch'>stats</a></td><td class='form'><form class='right' method='get' action='/aports/log/community/chromium/musl-sandbox.patch'>
<select name='qt'>
<option value='grep'>log msg</option>
<option value='author'>author</option>
<option value='committer'>committer</option>
<option value='range'>range</option>
</select>
<input class='txt' type='search' size='10' name='q' value=''/>
<input type='submit' value='search'/>
</form>
</td></tr></table>
<div class='path'>path: <a href='/aports/tree/'>root</a>/<a href='/aports/tree/community'>community</a>/<a href='/aports/tree/community/chromium'>chromium</a>/<a href='/aports/tree/community/chromium/musl-sandbox.patch'>musl-sandbox.patch</a></div><div class='content'>blob: 71af19fb88df40eece6cab868efbfd83480e944e (<a href='/aports/plain/community/chromium/musl-sandbox.patch'>plain</a>) (<a href='/aports/blame/community/chromium/musl-sandbox.patch'>blame</a>)
<table summary='blob content' class='blob'>
<tr><td class='linenumbers'><pre><a id='n1' href='#n1'>1</a>
<a id='n2' href='#n2'>2</a>
<a id='n3' href='#n3'>3</a>
<a id='n4' href='#n4'>4</a>
<a id='n5' href='#n5'>5</a>
<a id='n6' href='#n6'>6</a>
<a id='n7' href='#n7'>7</a>
<a id='n8' href='#n8'>8</a>
<a id='n9' href='#n9'>9</a>
<a id='n10' href='#n10'>10</a>
<a id='n11' href='#n11'>11</a>
<a id='n12' href='#n12'>12</a>
<a id='n13' href='#n13'>13</a>
<a id='n14' href='#n14'>14</a>
<a id='n15' href='#n15'>15</a>
<a id='n16' href='#n16'>16</a>
<a id='n17' href='#n17'>17</a>
<a id='n18' href='#n18'>18</a>
<a id='n19' href='#n19'>19</a>
<a id='n20' href='#n20'>20</a>
<a id='n21' href='#n21'>21</a>
<a id='n22' href='#n22'>22</a>
<a id='n23' href='#n23'>23</a>
<a id='n24' href='#n24'>24</a>
<a id='n25' href='#n25'>25</a>
<a id='n26' href='#n26'>26</a>
<a id='n27' href='#n27'>27</a>
<a id='n28' href='#n28'>28</a>
<a id='n29' href='#n29'>29</a>
<a id='n30' href='#n30'>30</a>
<a id='n31' href='#n31'>31</a>
<a id='n32' href='#n32'>32</a>
<a id='n33' href='#n33'>33</a>
<a id='n34' href='#n34'>34</a>
<a id='n35' href='#n35'>35</a>
<a id='n36' href='#n36'>36</a>
<a id='n37' href='#n37'>37</a>
<a id='n38' href='#n38'>38</a>
<a id='n39' href='#n39'>39</a>
<a id='n40' href='#n40'>40</a>
<a id='n41' href='#n41'>41</a>
<a id='n42' href='#n42'>42</a>
<a id='n43' href='#n43'>43</a>
<a id='n44' href='#n44'>44</a>
<a id='n45' href='#n45'>45</a>
<a id='n46' href='#n46'>46</a>
<a id='n47' href='#n47'>47</a>
<a id='n48' href='#n48'>48</a>
<a id='n49' href='#n49'>49</a>
<a id='n50' href='#n50'>50</a>
<a id='n51' href='#n51'>51</a>
<a id='n52' href='#n52'>52</a>
<a id='n53' href='#n53'>53</a>
<a id='n54' href='#n54'>54</a>
<a id='n55' href='#n55'>55</a>
<a id='n56' href='#n56'>56</a>
<a id='n57' href='#n57'>57</a>
<a id='n58' href='#n58'>58</a>
<a id='n59' href='#n59'>59</a>
<a id='n60' href='#n60'>60</a>
<a id='n61' href='#n61'>61</a>
<a id='n62' href='#n62'>62</a>
<a id='n63' href='#n63'>63</a>
<a id='n64' href='#n64'>64</a>
<a id='n65' href='#n65'>65</a>
<a id='n66' href='#n66'>66</a>
<a id='n67' href='#n67'>67</a>
<a id='n68' href='#n68'>68</a>
<a id='n69' href='#n69'>69</a>
<a id='n70' href='#n70'>70</a>
<a id='n71' href='#n71'>71</a>
<a id='n72' href='#n72'>72</a>
<a id='n73' href='#n73'>73</a>
<a id='n74' href='#n74'>74</a>
<a id='n75' href='#n75'>75</a>
<a id='n76' href='#n76'>76</a>
<a id='n77' href='#n77'>77</a>
<a id='n78' href='#n78'>78</a>
<a id='n79' href='#n79'>79</a>
<a id='n80' href='#n80'>80</a>
<a id='n81' href='#n81'>81</a>
<a id='n82' href='#n82'>82</a>
<a id='n83' href='#n83'>83</a>
<a id='n84' href='#n84'>84</a>
<a id='n85' href='#n85'>85</a>
<a id='n86' href='#n86'>86</a>
<a id='n87' href='#n87'>87</a>
<a id='n88' href='#n88'>88</a>
<a id='n89' href='#n89'>89</a>
<a id='n90' href='#n90'>90</a>
<a id='n91' href='#n91'>91</a>
<a id='n92' href='#n92'>92</a>
<a id='n93' href='#n93'>93</a>
<a id='n94' href='#n94'>94</a>
<a id='n95' href='#n95'>95</a>
<a id='n96' href='#n96'>96</a>
<a id='n97' href='#n97'>97</a>
<a id='n98' href='#n98'>98</a>
<a id='n99' href='#n99'>99</a>
<a id='n100' href='#n100'>100</a>
<a id='n101' href='#n101'>101</a>
<a id='n102' href='#n102'>102</a>
<a id='n103' href='#n103'>103</a>
<a id='n104' href='#n104'>104</a>
<a id='n105' href='#n105'>105</a>
<a id='n106' href='#n106'>106</a>
<a id='n107' href='#n107'>107</a>
<a id='n108' href='#n108'>108</a>
<a id='n109' href='#n109'>109</a>
<a id='n110' href='#n110'>110</a>
<a id='n111' href='#n111'>111</a>
<a id='n112' href='#n112'>112</a>
<a id='n113' href='#n113'>113</a>
<a id='n114' href='#n114'>114</a>
<a id='n115' href='#n115'>115</a>
<a id='n116' href='#n116'>116</a>
<a id='n117' href='#n117'>117</a>
<a id='n118' href='#n118'>118</a>
<a id='n119' href='#n119'>119</a>
<a id='n120' href='#n120'>120</a>
<a id='n121' href='#n121'>121</a>
<a id='n122' href='#n122'>122</a>
<a id='n123' href='#n123'>123</a>
<a id='n124' href='#n124'>124</a>
<a id='n125' href='#n125'>125</a>
<a id='n126' href='#n126'>126</a>
<a id='n127' href='#n127'>127</a>
<a id='n128' href='#n128'>128</a>
<a id='n129' href='#n129'>129</a>
<a id='n130' href='#n130'>130</a>
<a id='n131' href='#n131'>131</a>
<a id='n132' href='#n132'>132</a>
<a id='n133' href='#n133'>133</a>
<a id='n134' href='#n134'>134</a>
<a id='n135' href='#n135'>135</a>
<a id='n136' href='#n136'>136</a>
<a id='n137' href='#n137'>137</a>
<a id='n138' href='#n138'>138</a>
<a id='n139' href='#n139'>139</a>
<a id='n140' href='#n140'>140</a>
<a id='n141' href='#n141'>141</a>
<a id='n142' href='#n142'>142</a>
<a id='n143' href='#n143'>143</a>
<a id='n144' href='#n144'>144</a>
<a id='n145' href='#n145'>145</a>
<a id='n146' href='#n146'>146</a>
<a id='n147' href='#n147'>147</a>
<a id='n148' href='#n148'>148</a>
<a id='n149' href='#n149'>149</a>
<a id='n150' href='#n150'>150</a>
<a id='n151' href='#n151'>151</a>
<a id='n152' href='#n152'>152</a>
<a id='n153' href='#n153'>153</a>
<a id='n154' href='#n154'>154</a>
<a id='n155' href='#n155'>155</a>
<a id='n156' href='#n156'>156</a>
<a id='n157' href='#n157'>157</a>
<a id='n158' href='#n158'>158</a>
</pre></td>
<td class='lines'><pre><code><span style="color:#000000; font-weight:bold">--- ./sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc</span>
<span style="color:#0057ae">+++ ./sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc</span>
<span style="color:#010181">&#64;&#64; -139,21 +139,11 &#64;&#64;</span>
 // present (as in newer versions of posix_spawn).
 ResultExpr RestrictCloneToThreadsAndEPERMFork() {
   const Arg&lt;unsigned long&gt; flags(0);
<span style="color:#000000; font-weight:bold">-</span>
<span style="color:#000000; font-weight:bold">-  // TODO(mdempsky): Extend DSL to support (flags &amp; ~mask1) == mask2.</span>
<span style="color:#000000; font-weight:bold">-  const uint64_t kAndroidCloneMask = CLONE_VM | CLONE_FS | CLONE_FILES |</span>
<span style="color:#000000; font-weight:bold">-                                     CLONE_SIGHAND | CLONE_THREAD |</span>
<span style="color:#000000; font-weight:bold">-                                     CLONE_SYSVSEM;</span>
<span style="color:#000000; font-weight:bold">-  const uint64_t kObsoleteAndroidCloneMask = kAndroidCloneMask | CLONE_DETACHED;</span>
<span style="color:#000000; font-weight:bold">-</span>
<span style="color:#000000; font-weight:bold">-  const uint64_t kGlibcPthreadFlags =</span>
<span style="color:#000000; font-weight:bold">-      CLONE_VM | CLONE_FS | CLONE_FILES | CLONE_SIGHAND | CLONE_THREAD |</span>
<span style="color:#000000; font-weight:bold">-      CLONE_SYSVSEM | CLONE_SETTLS | CLONE_PARENT_SETTID | CLONE_CHILD_CLEARTID;</span>
<span style="color:#000000; font-weight:bold">-  const BoolExpr glibc_test = flags == kGlibcPthreadFlags;</span>
<span style="color:#000000; font-weight:bold">-</span>
<span style="color:#000000; font-weight:bold">-  const BoolExpr android_test =</span>
<span style="color:#000000; font-weight:bold">-      AnyOf(flags == kAndroidCloneMask, flags == kObsoleteAndroidCloneMask,</span>
<span style="color:#000000; font-weight:bold">-            flags == kGlibcPthreadFlags);</span>
<span style="color:#0057ae">+  const int required = CLONE_VM | CLONE_FS | CLONE_FILES | CLONE_SIGHAND |</span>
<span style="color:#0057ae">+                       CLONE_THREAD | CLONE_SYSVSEM;</span>
<span style="color:#0057ae">+  const int safe = CLONE_SETTLS | CLONE_PARENT_SETTID | CLONE_CHILD_CLEARTID |</span>
<span style="color:#0057ae">+                   CLONE_DETACHED;</span>
<span style="color:#0057ae">+  const BoolExpr thread_clone_ok = (flags&amp;~safe)==required;</span>
 
   // The following two flags are the two important flags in any vfork-emulating
   // clone call. EPERM any clone call that contains both of them.
<span style="color:#010181">&#64;&#64; -163,7 +153,7 &#64;&#64;</span>
       AnyOf((flags &amp; (CLONE_VM | CLONE_THREAD)) == 0,
             (flags &amp; kImportantCloneVforkFlags) == kImportantCloneVforkFlags);
 
<span style="color:#000000; font-weight:bold">-  return If(IsAndroid() ? android_test : glibc_test, Allow())</span>
<span style="color:#0057ae">+  return If(thread_clone_ok, Allow())</span>
       .ElseIf(is_fork_or_clone_vfork, Error(EPERM))
       .Else(CrashSIGSYSClone());
 }
<span style="color:#000000; font-weight:bold">--- ./sandbox/linux/seccomp-bpf-helpers/syscall_sets.cc</span>
<span style="color:#0057ae">+++ ./sandbox/linux/seccomp-bpf-helpers/syscall_sets.cc</span>
<span style="color:#010181">&#64;&#64; -392,6 +392,7 &#64;&#64;</span>
 #if defined(__i386__)
     case __NR_waitpid:
 #endif
<span style="color:#0057ae">+    case __NR_set_tid_address:</span>
       return true;
     case __NR_clone:  // Should be parameter-restricted.
     case __NR_setns:  // Privileged.
<span style="color:#010181">&#64;&#64; -404,7 +405,6 &#64;&#64;</span>
 #if defined(__i386__) || defined(__x86_64__) || defined(__mips__)
     case __NR_set_thread_area:
 #endif
<span style="color:#000000; font-weight:bold">-    case __NR_set_tid_address:</span>
     case __NR_unshare:
 #if !defined(__mips__) &amp;&amp; !defined(__aarch64__)
     case __NR_vfork:
<span style="color:#010181">&#64;&#64; -514,6 +514,8 &#64;&#64;</span>
     case __NR_mlock:
     case __NR_munlock:
     case __NR_munmap:
<span style="color:#0057ae">+    case __NR_mremap:</span>
<span style="color:#0057ae">+    case __NR_membarrier:</span>
       return true;
     case __NR_madvise:
     case __NR_mincore:
<span style="color:#010181">&#64;&#64; -531,7 +533,6 &#64;&#64;</span>
     case __NR_modify_ldt:
 #endif
     case __NR_mprotect:
<span style="color:#000000; font-weight:bold">-    case __NR_mremap:</span>
     case __NR_msync:
     case __NR_munlockall:
     case __NR_readahead:
<span style="color:#000000; font-weight:bold">--- ./sandbox/linux/system_headers/arm64_linux_syscalls.h</span>
<span style="color:#0057ae">+++ ./sandbox/linux/system_headers/arm64_linux_syscalls.h</span>
<span style="color:#010181">&#64;&#64; -1063,4 +1063,8 &#64;&#64;</span>
 #define __NR_memfd_create 279
 #endif
 
<span style="color:#0057ae">+#if !defined(__NR_membarrier)</span>
<span style="color:#0057ae">+#define __NR_membarrier 283</span>
<span style="color:#0057ae">+#endif</span>
<span style="color:#0057ae">+</span>
 #endif  // SANDBOX_LINUX_SYSTEM_HEADERS_ARM64_LINUX_SYSCALLS_H_
<span style="color:#000000; font-weight:bold">--- ./sandbox/linux/system_headers/arm_linux_syscalls.h</span>
<span style="color:#0057ae">+++ ./sandbox/linux/system_headers/arm_linux_syscalls.h</span>
<span style="color:#010181">&#64;&#64; -1385,6 +1385,10 &#64;&#64;</span>
 #define __NR_memfd_create (__NR_SYSCALL_BASE+385)
 #endif
 
<span style="color:#0057ae">+#if !defined(__NR_membarrier)</span>
<span style="color:#0057ae">+#define __NR_membarrier (__NR_SYSCALL_BASE+389)</span>
<span style="color:#0057ae">+#endif</span>
<span style="color:#0057ae">+</span>
 // ARM private syscalls.
 #if !defined(__ARM_NR_BASE)
 #define __ARM_NR_BASE (__NR_SYSCALL_BASE + 0xF0000)
<span style="color:#000000; font-weight:bold">--- ./sandbox/linux/system_headers/mips64_linux_syscalls.h</span>
<span style="color:#0057ae">+++ ./sandbox/linux/system_headers/mips64_linux_syscalls.h</span>
<span style="color:#010181">&#64;&#64; -1271,4 +1271,8 &#64;&#64;</span>
 #define __NR_memfd_create (__NR_Linux + 314)
 #endif
 
<span style="color:#0057ae">+#if !defined(__NR_membarrier)</span>
<span style="color:#0057ae">+#define __NR_membarrier (__NR_Linux  318)</span>
<span style="color:#0057ae">+#endif</span>
<span style="color:#0057ae">+</span>
 #endif  // SANDBOX_LINUX_SYSTEM_HEADERS_MIPS64_LINUX_SYSCALLS_H_
<span style="color:#000000; font-weight:bold">--- ./sandbox/linux/system_headers/mips_linux_syscalls.h</span>
<span style="color:#0057ae">+++ ./sandbox/linux/system_headers/mips_linux_syscalls.h</span>
<span style="color:#010181">&#64;&#64; -1433,4 +1433,8 &#64;&#64;</span>
 #define __NR_memfd_create (__NR_Linux + 354)
 #endif
 
<span style="color:#0057ae">+#if !defined(__NR_membarrier)</span>
<span style="color:#0057ae">+#define __NR_membarrier (__NR_Linux  358)</span>
<span style="color:#0057ae">+#endif</span>
<span style="color:#0057ae">+</span>
 #endif  // SANDBOX_LINUX_SYSTEM_HEADERS_MIPS_LINUX_SYSCALLS_H_
<span style="color:#000000; font-weight:bold">--- ./sandbox/linux/system_headers/x86_64_linux_syscalls.h</span>
<span style="color:#0057ae">+++ ./sandbox/linux/system_headers/x86_64_linux_syscalls.h</span>
<span style="color:#010181">&#64;&#64; -1290,5 +1290,9 &#64;&#64;</span>
 #define __NR_memfd_create 319
 #endif
 
<span style="color:#0057ae">+#if !defined(__NR_membarrier)</span>
<span style="color:#0057ae">+#define __NR_membarrier 324</span>
<span style="color:#0057ae">+#endif</span>
<span style="color:#0057ae">+</span>
 #endif  // SANDBOX_LINUX_SYSTEM_HEADERS_X86_64_LINUX_SYSCALLS_H_
 
<span style="color:#000000; font-weight:bold">--- ./services/service_manager/sandbox/linux/bpf_renderer_policy_linux.cc</span>
<span style="color:#0057ae">+++ ./services/service_manager/sandbox/linux/bpf_renderer_policy_linux.cc</span>
<span style="color:#010181">&#64;&#64; -88,10 +88,10 &#64;&#64;</span>
     case __NR_sysinfo:
     case __NR_times:
     case __NR_uname:
<span style="color:#000000; font-weight:bold">-      return Allow();</span>
<span style="color:#000000; font-weight:bold">-    case __NR_sched_getaffinity:</span>
     case __NR_sched_getparam:
     case __NR_sched_getscheduler:
<span style="color:#0057ae">+      return Allow();</span>
<span style="color:#0057ae">+    case __NR_sched_getaffinity:</span>
     case __NR_sched_setscheduler:
       return sandbox::RestrictSchedTarget(GetPolicyPid(), sysno);
     case __NR_prlimit64:
diff --git a/sandbox/linux/system_headers/linux_syscalls.h b/sandbox/linux/system_headers/linux_syscalls.h
index 2b78a0c..b6fedb5 100644
<span style="color:#000000; font-weight:bold">--- ./sandbox/linux/system_headers/linux_syscalls.h</span>
<span style="color:#0057ae">+++ ./sandbox/linux/system_headers/linux_syscalls.h</span>
<span style="color:#010181">&#64;&#64; -10,6 +10,7 &#64;&#64;</span>
 #define SANDBOX_LINUX_SYSTEM_HEADERS_LINUX_SYSCALLS_H_
 
 #include &quot;build/build_config.h&quot;
<span style="color:#0057ae">+#include &lt;sys/syscall.h&gt;</span>
 
 #if defined(__x86_64__)
 #include &quot;sandbox/linux/system_headers/x86_64_linux_syscalls.h&quot;
</code></pre></td></tr></table>
</div> <!-- class=content -->
<div class='footer'>generated by <a href='https://git.zx2c4.com/cgit/about/'>cgit v1.2.1</a> (<a href='https://git-scm.com/'>git 2.18.0</a>) at 2020-06-06 19:56:33 +0000</div>
</div> <!-- id=cgit -->
</body>
</html>
